src/SharedUtilities/Domains.jl

```diff
@@ -281,8 +253,65 @@ function coordinates(
     return cc
 end
 
+
+### Example of component specific domain
+"""
+    AbstractVegetationDomain{FT} <: AbstractDomain{FT}
+
+An abstract type for vegetation specific domains.
+"""
+abstract type AbstractVegetationDomain{FT} <: AbstractDomain{FT} end
+
+
+"""
+   RootDomain{FT} <: AbstractVegetationDomain{FT}
+
+Domain for a single bulk plant with roots of varying depths. The user needs
+to specify the depths of the root tips as wel as the heights of the
+compartments to be modeled within the plant. The compartment heights
+are expected to be sorted in ascending order.
+"""
+struct RootDomain{FT} <: AbstractVegetationDomain{FT}
+    "The depth of the root tips, in meters"
+    root_depths::Vector{FT}
+    "The height of the stem, leaf compartments, in meters"
+    compartment_heights::Vector{FT}
+end
+
+function coordinates(domain::RootDomain{FT}) where {FT}
+    return domain.compartment_heights
+end
+
+
+struct SingleColumnLSM{FT} <: AbstractDomain{FT}
+    subsurface::Column{FT}
+    surface::Point{FT}
+end
+
+function SingleColumnLSM(
+    FT::DataType = Float64;
+    zlim::Tuple{G, G},
```

Comment:
hmm yes, that would seem clearer. not sure why I did it this way.

also, is this function type stable (FT is an argument, depending on its value, the type of the output changes), even if we change it to what you suggested? perhaps need to rewrite a bit also?

Reply:
maybe we always need FT to be passed as parametric type, also for the `Column` outer constructor.

Reply:
Yes, this is type stable since the types of the input determine the output type (and the types of the other internal variables). It doesn't matter that `FT` is ambiguous when writing the code. I think that under the hood Julia would compile a method for each `FT` that is actually used in the driver. But yes, it can compile efficient code here.

Reply:
thanks for the explanation!